app/controllers/registration_commands_controller.rb

```diff
@@ -44,6 +44,11 @@ def registration_params
     end
   end
 
+  def host_groups
+    Hostgroup.authorized(:view_hostgroups)
+             .map { |hg| hg.as_json.merge({'inherited_operatingsystem_id': hg.os&.id}) }
```

Comment:
Isn't `os` part of `hg.as_json`? Or should `os` simply be included altogether rather than adding a new field `inherited_operatingsystem_id`?

Reply:
> Isn't os part of hg.as_json?

No, only the `operatingsystem_id`, which returns only id of os assigned directly to the host group (ignoring parent host group)

> Or should os simply be included altogether

TBH I don't know how to do it.

Reply:
Perhaps this is starting to be complex enough that RABL should be used? @ares any thoughts?

Reply:
> Perhaps this is starting to be complex enough that RABL should be used?

Yes we could do it (and probably we should), but that would require also changes in the Katello because we extend `plugin_data` [0]. That would complicate the cherry-picking and getting the fix to F2.5 and Sat 6.10.

I created Redmine tasks [1], would you be okay with merging this fix without the RABL stuff, so we can get it easily to F2.5 & Sat 6.10?

[0] [katello/concerns/registration_commands_controller_extensions.rb](https://github.com/Katello/katello/blob/master/app/controllers/katello/concerns/registration_commands_controller_extensions.rb)
[1] [#32734](https://projects.theforeman.org/issues/32734) & [#32735](https://projects.theforeman.org/issues/32735)

Reply:
It is a good point, I'd keep it simple for this fix, we can move to RABL in separate PR :)

Reply:
With that I think this thread is resolved. See https://github.com/theforeman/foreman/pull/8534/files#r646469041 for a good suggestion to solve it right now.